rust/toolchain.bzl

```diff
@@ -75,6 +76,14 @@ def _rust_stdlib_filegroup_impl(ctx):
                 print("File partitioned: {}".format(f.basename))
             fail("rust_toolchain couldn't properly partition rlibs in rust_std. Partitioned {} out of {} files. This is probably a bug in the rule implementation.".format(partitioned_files_len, len(dot_a_files)))
 
+    std_dylib = None
+
+    for file in rust_std:
+        # find libstd-*.so
+        if file.basename.startswith("libstd-") and file.basename.endswith(".so"):
```

Comment:
It feels like we should probably extract out this check to a function somewhere common with `is_std_dylib` in rust/private/utils.bzl, and maybe add a `or file.basename.endswith(".dll")` case for Windows?

Reply:
Yes. I think this is why the tests are failing.